x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/TransformIndexerStateTests.java

```diff
@@ -684,6 +674,132 @@ public void testStopAtCheckpoint() throws Exception {
+         * When a user calls _stop?force=false
+         * Then the indexer thread should exit early
+         */
+        {
+            var indexer = createMockIndexer(
+                createTransformConfig(),
+                new AtomicReference<>(IndexerState.STARTED),
+                null,
+                threadPool,
+                auditor,
+                null,
+                new TransformIndexerStats(),
+                new TransformContext(TransformTaskState.STARTED, "", 0, mock(TransformContext.Listener.class))
+            );
+
+            // stop the indexer thread once it kicks off
+            var startLatch = indexer.createAwaitForStartLatch(1);
+            assertEquals(IndexerState.STARTED, indexer.start());
+            assertTrue(indexer.maybeTriggerAsyncJob(System.currentTimeMillis()));
+            assertEquals(IndexerState.INDEXING, indexer.getState());
+
+            // stop the indexer, equivalent to _stop?force=false
+            assertEquals(IndexerState.STOPPING, indexer.stop());
+            assertEquals(IndexerState.STOPPING, indexer.getState());
+
+            // now let the indexer thread run
+            startLatch.countDown();
+
+            assertBusy(() -> {
+                assertThat(indexer.getState(), equalTo(IndexerState.STOPPED));
+                assertThat(indexer.getLastCheckpoint().getCheckpoint(), equalTo(-1L));
+            });
+        }
+
+        /*
+         * Given a started transform
+         * And the indexer thread has not started yet
+         * When a user calls _stop?force=true
+         * Then the indexer thread should exit early
+         */
+        {
+            var indexer = createMockIndexer(
+                createTransformConfig(),
+                new AtomicReference<>(IndexerState.STARTED),
+                null,
+                threadPool,
+                auditor,
+                null,
+                new TransformIndexerStats(),
+                new TransformContext(TransformTaskState.STARTED, "", 0, mock(TransformContext.Listener.class))
+            );
+
+            // stop the indexer thread once it kicks off
+            var startLatch = indexer.createAwaitForStartLatch(1);
+            assertEquals(IndexerState.STARTED, indexer.start());
+            assertTrue(indexer.maybeTriggerAsyncJob(System.currentTimeMillis()));
+            assertEquals(IndexerState.INDEXING, indexer.getState());
+
+            // stop the indexer, equivalent to _stop?force=true
+            assertFalse("Transform Indexer thread should still be running", indexer.abort());
+            assertEquals(IndexerState.ABORTING, indexer.getState());
+
+            // now let the indexer thread run
+            startLatch.countDown();
+
+            assertBusy(() -> {
+                assertThat(indexer.getState(), equalTo(IndexerState.ABORTING));
+                assertThat(indexer.getLastCheckpoint().getCheckpoint(), equalTo(-1L));
+            });
+        }
+
+        /*
```

Comment:
It seems there are 3 distinct test cases in this one test method. Could you split this method into 3 (one per logical test case)?